src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -839,6 +839,18 @@ private void CrashGracefully(Exception ex, bool fatal = false)
         {
             try
             {
+                if (!fatal)
+                {
+                    var localPkg =  Model.GetPackageManagerExtension()?.PackageLoader?.LocalPackages?.FirstOrDefault(p => p.LoadedAssemblies.FirstOrDefault(a => a.Name == ex.Source) != null);
```

Comment:
might happen that one of these package assemblies is already loaded by dynamo (as a dependency), so we could potentially continue running even though the exception is coming from somewhere else. (not the pacakge we identified)
But I think that might be an edge case.

Reply:
At the moment this code will always continue the session and only log the exception. 
As an alternative, we could show a warning message that exception ex was thrown from package X and give the user a choice to continue the dynamo seasion or crash and send CER report 
@Amoursol any preferences?

Reply:
`If the source property is not set explicitly, the runtime automatically sets it to the name of the assembly in which the exception originated.`

hmm, so in most cases this will work.

Have you verified that this property is set to the short name or full name of the assembly? Does it include the version number and culture?

Reply:
what about the edge case where a user has shipped system binaries (Microsoft) or Dynamo binaries?

Reply:
what about also logging something to ADP so we have analytics on how often these crashes occur without actually crashing?

Reply:
`Have you verified that this property is set to the short name or full name of the assembly`
It is just the short name. Obviously this can be overridden by custom exceptions.

Reply:
`what about the edge case where a user has shipped system binaries (Microsoft) or Dynamo binaries?`
Not sure at the moment how I can identify these sort of cases...Are the package `LoadedAssemblies` actually loaded or they can just be on disk without actually being the ones to be loaded...I can check this

Reply:
Yea, `LoadedAssemblies` could be already loaded assemblies. So if a user supplies a dotnet system dll, then all the exceptions from that dll (even if it was loaded by dotnet previously) will not cause a hard crash anymore.

I can verify if the assemblies have been loaded already, probably at a small performance penalty

Reply:
Minor: does it make sense to have this check (code path) in the same function (`CrashGracefully`), as it isn't crashing in this case?

Reply:
CrashGracefully is called from multiple places.
Another way I could go is to not subscribe to the unhandled_exception_handlers at all during testing.